documentation/api.tex

```diff
@@ -0,0 +1,69 @@
+The data in the middle column is from the query login. This shows that it only returns exactly the 2 parameters i asked for when i queried.
+
+\graphic{1}{api/graphiql}{The GraphiQL interface in a browser}{fig:graphiql}
+
+\section{Root}
+On figure \ref{fig:root} the two root types are shown. 
+These are Query and Mutation. 
+Query can be considered as GET requests and Mutations can be considered as POST requests.
+\graphich{0.6}{api/rootType}{Root types}{fig:root}
+
+\section{Query}
+On figure \ref{fig:query} the \verb+Query+ root type is shown. It contains two fields. 
+
+The first field \verb+viewer+ is where the restricted data is located.
+It takes a token string, and returns a viewer type which is shown on figure \ref{fig:viewerType}.
+
+The second field \verb+login+ also returns a viewer type. 
+This is to save a round trip.
+The viewer type has a \verb+token+ field that can be used to query the \verb+viewer+ field at a later time after logging in.
+\graphich{0.6}{api/queryType}{The Query root type with fields}{fig:query}
+
+\section{Viewer}
+The \verb+Viewer+ can only be retrieved if the user is authenticated. 
+Either through the \verb+viewer+ field or \verb+login+ field on the root type \verb+query+. The \verb+Viewer+ is shown on figure \ref{fig:viewerType}
+\graphich{0.6}{api/viewerType}{The Query root type with fields}{fig:viewerType}
+
+\section{Mutation}
+The \verb+Mutation+ root type is also a restricted field where the user needs to be authenticated. 
+It takes a \verb+token+ string which is required. 
+Required is shown by the ! after String. 
+It returns a MutationViewer that is shown on figure \ref{fig:mutationViewer}
+\graphich{0.6}{api/mutationType}{The Mutation root type with fields}{fig:mutationType}
+
+\section{MutationViewer}
+The \verb+MutationViewer+ has 3 functions that takes parameters and a logout parameter that just needs to be required on the MutationViewer to be called. The \verb+MutationViewer+ can be seen on figure \ref{fig:mutationViewer}
+\graphich{0.6}{api/mutationViewerType}{The MutationViewer type with fields}{fig:mutationViewer}
+
+\section{CreateUser}
+The \verb+createUser+ function is used to create users with minimal information. 
+As shown on figure \ref{fig:createUser}, it only takes a \verb+Email+ and a \verb+Password+.
+\graphich{0.6}{api/createUserType}{The MutationViewer type with fields}{fig:createUser}
+
+\section{CreateAutoUser}
+The \verb+createAutoUser+ function can be used by consultants and coaches to create a user with an autogenerated password.
+The user then receives the password on the provided email.
+It takes a \verb+AutoUser+ as an input parameter.
+\graphich{0.6}{api/createAutoUserType}{The MutationViewer type with fields}{fig:mutationViewer}
+
+\section{AutoUser}
+When functions take several fields, an input type can be created. AutoUser is an input type used in \verb+CreateAutoUser+ and can be seen on figure \ref{fig:autoUser}.
```

Comment:
I think you can leave out the first sentence. It is GraphQL explanatory more than documentation.
verb on AutoUser